### PR TITLE
[WIP] reduce the number of block hashes

### PR DESCRIPTION
**Pull request by songgao**
An attempt to [KBFSKBFS-1156](https://keybase.atlassian.net/browse/KBFS-1156)

### CODE REVIEW

**Comment by songgao** [comment]
Here's some `cpuprofile` entries:

Master:

```
$ go tool pprof -top /Users/songgao/gopath/src/github.com/keybase/kbfs/kbfsfuse/kbfsfuse profile-master | grep sha256
     0.19s  7.34% 79.54%      0.19s  7.34%  crypto/sha256.block
         0     0% 94.59%      0.27s 10.42%  crypto/sha256.(*digest).Write
         0     0% 94.59%      0.27s 10.42%  crypto/sha256.Sum256
```

This PR:

```
$ go tool pprof -top /Users/songgao/gopath/src/github.com/keybase/kbfs/kbfsfuse/kbfsfuse profile-patch | grep sha256
     0.15s  5.84% 82.10%      0.15s  5.84%  crypto/sha256.block
         0     0% 96.11%      0.20s  7.78%  crypto/sha256.(*digest).Write
         0     0% 96.11%      0.20s  7.78%  crypto/sha256.Sum256
```

**Comment by strib** [comment]
Nice, 21% improvement with just a few lines of code!

File: libkbfs/block_types.go [hunk 1]
```diff
@@ -99,6 +99,8 @@ type FileBlock struct {
 	Contents []byte `codec:"c,omitempty"`
 	// if indirect, contains the indirect pointers to the next level of blocks
 	IPtrs []IndirectFilePtr `codec:"i,omitempty"`
+
+	hash *RawDefaultHash
```
Comment (strib) [inline libkbfs/block_types.go]: Please add a comment above this explaining that this is a hash of the _plaintext_ contents of the block, and is only set on direct file blocks.

**Comment by strib** [comment]
Looks good to me, besides one nit.  Merge when ready!

**Comment by songgao** [comment]
Thanks! I'll add the comment and rebase before merging.

**Pull request merged**